spec/integration/api/users_controller_spec.rb

```diff
@@ -0,0 +1,496 @@
+# frozen_sTring_literal: true
+
+require "spec_helper"
+
+describe Api::V1::UsersController do
+  include PeopleHelper
+
+  let(:auth) { FactoryGirl.create(:auth_with_read_and_write) }
+  let(:auth_read_only) { FactoryGirl.create(:auth_with_read) }
+  let!(:access_token) { auth.create_access_token.to_s }
+  let!(:access_token_read_only) { auth_read_only.create_access_token.to_s }
+
+  describe "#show" do
+    context "Current User" do
+      it "succeeds when logged in" do
+        get(
+          api_v1_user_path,
+          params: {access_token: access_token}
+        )
+        user = JSON.parse(response.body)
+        expect(response.status).to eq(200)
+        expect(user["guid"]).to eq(auth.user.guid)
+        confirm_self_data_format(user)
+      end
+
+      it "fails if invalid token" do
+        get(
+          api_v1_user_path,
+          params: {access_token: "999_999_999"}
+        )
+        expect(response.status).to eq(401)
+      end
+    end
+
+    context "Single User" do
+      it "succeeds with public user not in Aspect" do
+        alice.profile[:public_details] = true
+        alice.profile.save
+        get(
+          "/api/v1/users/#{alice.guid}",
```

Comment:
Why don't you use Rails path helpers here?

Reply:
I couldn't get them to generate that path no matter how hard I tried.  It happened on another endpoint as well.